### PR TITLE
Wms get map pdf

### PR DESCRIPTION
**Pull request by mhugent**
PDF output for QGIS Server WMS is currently only available in the GetPrint request. This PR implements support for pdf output to the GetMap request

### CODE REVIEW

File: src/server/services/wms/qgspdfwriter.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#ifndef QGSPDFWRITER_H
+#define QGSPDFWRITER_H
+
+#include "qgswmsrequest.h"
+
+namespace QgsWms
+{
+
+  /**
+   * Output GetMap response in DXF format
```
Comment (m-kuhn) [inline src/server/services/wms/qgspdfwriter.h:26]: Leftover?

File: src/server/services/wms/qgswmsrenderer.cpp [hunk 2]
```diff
@@ -1133,6 +1134,33 @@ namespace QgsWms
...
+      pdfExportDetails.producer = QStringLiteral( "QGIS %1" ).arg( Qgis::version() );
+      pdfExportDetails.creator = QStringLiteral( "QGIS %1" ).arg( Qgis::version() );
+      pdfExportDetails.creationDateTime = QDateTime::currentDateTime();
+      pdfExportDetails.subject = QgsProject::instance()->metadata().abstract();
+      pdfExportDetails.title = QgsProject::instance()->metadata().title();
+      pdfExportDetails.keywords = QgsProject::instance()->metadata().keywords();
+    }
+    pdfExportDetails.useIso32000ExtensionFormatGeoreferencing = mWmsParameters.pdfUseIso32000ExtensionFormatGeoreferencing();
+    pdfExportDetails.useOgcBestPracticeFormatGeoreferencing = mWmsParameters.pdfUseOgcBestPracticeFormatGeoreferencing();
+    bool geoPdf = mWmsParameters.pdfAppendGeoreference();
```
Comment (pblottiere) [inline src/server/services/wms/qgswmsrenderer.cpp:1159]:
```suggestion
    const bool geoPdf = mWmsParameters.pdfAppendGeoreference();
```

File: src/server/services/wms/qgswmsrenderer.h [hunk 3]
```diff
@@ -130,6 +131,13 @@ namespace QgsWms
       */
       std::unique_ptr<QgsDxfExport> getDxf();
 
+      /**
+       * Returns a configured pdf export task
+       * \tmpFileName the name of the temporary file to store the pdf
+       * \returns pdf export object
```
Comment (pblottiere) [inline src/server/services/wms/qgswmsrenderer.h:137]:
```suggestion
       * \returns pdf export object
       * \since QGIS 3.36
```

**Comment by pblottiere** [comment]
Hello @mhugent,

Can you add a unit test please?

**Review by manisandro — APPROVED** [review]

**Pull request merged**